### PR TITLE
Feature/command improvements

### PR DESCRIPTION
**Pull request by dantleech**
This PR contains 3 commits which fix some issues in the user/role commands

- Handle the case where either a role or a user already exists
- Handle the case where the requested role does not exist (non-interactive)
- Indicate which roles *are* available in the Exception message (non-interactive)
- Handle the case where no roles exist (interactive)
- Added `preUpdate` and `prePersist` hooks to set the created + changed fields on the `Role` entity.

__Tasks:__

- [x] test coverage
- [ ] gather feedback for my changes
- [ ] added changelog line

__Informations:__

| Q             | A
| ------------- | ---
| Tests pass?   | yes
| Fixed tickets | #683 ]
| BC Breaks     | NO
| Doc           | NO

### CODE REVIEW

File: src/Sulu/Bundle/SecurityBundle/Command/CreateRoleCommand.php [hunk 1]
```diff
@@ -43,13 +43,25 @@ protected function execute(InputInterface $input, OutputInterface $output)
     {
         $doctrine = $this->getContainer()->get('doctrine');
         $em = $doctrine->getManager();
-        $now = new \Datetime();
```
Comment (dantleech) [inline src/Sulu/Bundle/SecurityBundle/Command/CreateRoleCommand.php]: Handled by Doctrine event hooks.

File: src/Sulu/Bundle/SecurityBundle/Command/CreateUserCommand.php [hunk 2]
```diff
@@ -220,22 +236,13 @@ function ($locale) use ($doctrine) {
...
-            $query = $doctrine->getRepository('SuluSecurityBundle:Role')
-                ->createQueryBuilder('role')
-                ->select('role.name')
-                ->getQuery();
-
-            $roles = array();
-            foreach ($query->getArrayResult() as $roleEntity) {
-                $roles[] = $roleEntity['name'];
-            }
+            $roleNames = $this->getRoleNames();
```
Comment (dantleech) [inline src/Sulu/Bundle/SecurityBundle/Command/CreateUserCommand.php]: Moved this to repository.

File: src/Sulu/Bundle/SecurityBundle/Resources/config/doctrine/BaseRole.orm.xml [hunk 3]
```diff
@@ -28,5 +28,10 @@
             </join-columns>
         </many-to-one>
 
+        <lifecycle-callbacks>
+            <lifecycle-callback type="prePersist" method="beforeSave" />
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/SecurityBundle/Resources/config/doctrine/BaseRole.orm.xml]: @dantleech why call the method like event?

Comment (dantleech) [inline src/Sulu/Bundle/SecurityBundle/Resources/config/doctrine/BaseRole.orm.xml]: updateTimestamps

**Pull request merged**